src/main/java/io/split/android/client/service/sseclient/notifications/NotificationProcessor.java

```diff
@@ -59,15 +71,31 @@ public void process(String rawJson) {
     }
 
     private void processSplitUpdate(SplitsChangeNotification notification) {
-
+        mSplitsUpdateNotificationsQueue.offer(notification);
     }
 
     private void processSplitKill(SplitKillNotification notification) {
-
+        ParameterizableSplitTask<Split> task = mSplitTaskFactory.createSplitKillTask();
```

Comment:
remember that we have to update the `changeNumber` in the Split storage, if the Kill notification is newer. As the `SplitKillTask` is not doing it (via the `SplitsStorage::updateWithoutChecks` method), I guess you will do it in `Synchronizer:: synchronizeSplits` method.